include/seqan/blast/blast_base.h

```diff
@@ -0,0 +1,669 @@
+constexpr BlastProgram
+getBlastProgram(BlastProgram const, BlastProgramTag<p> const &)
+{
+    return p;
+}
+
+// this will likely be evaluated at run-time
+constexpr BlastProgram
+getBlastProgram(BlastProgram const p, BlastProgramTag<BlastProgram::UNKNOWN> const &)
+{
+    return p;
+}
+
+// ----------------------------------------------------------------------------
+// qHasRevComp()
+// ----------------------------------------------------------------------------
+
+/*!
+ * @fn BlastProgramTag#qHasRevComp
+ * @brief Whether the reverse complement of the <b>query</b> sequence(s) is searched with the given BlastProgram
+ * @signature constexpr bool qHasRevComp(BlastProgram const p);
+ * template <BlastProgram p>
+ * constexpr bool qHasRevComp(BlastProgramTag<p> const &);
+ * template <BlastProgram p>
+ * constexpr bool qHasRevComp(BlastProgram const _p, BlastProgramTag<p> const &);
+ *
+ * @section Remarks
+ *
+ * This function can be used to check whether the defined BlastProgam implies the existence/creation/... of a
+ * reverse complement sequence of the query sequence.
+ *
+ * If both a tag and an enum are specified, the tag is given preference, unless it is set to
+ * @link BlastProgram::UNKNOWN @endlink.
+ *
+ * @return false for @link BlastProgram::BLASTP @endlink and @link BlastProgram::TBLASTN @endlink
+ * @return true otherwise
+ * @headerfile <seqan/blast.h>
+ */
+
+constexpr bool
+qHasRevComp(BlastProgram const p)
+{
+    return ((p==BlastProgram::BLASTP) || (p==BlastProgram::TBLASTN))
+            ? false
+            : true;
+}
+
+template <BlastProgram p>
+constexpr bool
+qHasRevComp(BlastProgramTag<p> const &)
+{
+    return qHasRevComp(p);
+}
+
+template <BlastProgram p>
+constexpr bool
+qHasRevComp(BlastProgram const _p, BlastProgramTag<p> const &)
+{
+    return qHasRevComp(getBlastProgram(_p, BlastProgramTag<p>()));
+}
+
+// ----------------------------------------------------------------------------
+// qIsTranslated()
+// ----------------------------------------------------------------------------
+
+/*!
+ * @fn BlastProgramTag#qIsTranslated
+ * @brief Whether the <b>query</b> sequence is translated in the given BlastProgram
+ * @signature constexpr bool qIsTranslated(BlastProgram const p);
+ * template <BlastProgram p>
+ * constexpr bool qIsTranslated(BlastProgramTag<p> const &);
+ * template <BlastProgram p>
+ * constexpr bool qIsTranslated(BlastProgram const _p, BlastProgramTag<p> const &);
+ *
+ * @section Remarks
+ *
+ * This function can be used to check whether the defined BlastProgam implies the
+ * nucleotide to aminoacid translation of the query sequence.
+ *
+ * If both a tag and an enum are specified, the tag is given preference, unless it is set to
+ * @link BlastProgram::UNKNOWN @endlink.
+ *
+ * @return true for @link BlastProgram::BLASTX @endlink and @link BlastProgram::TBLASTX @endlink
+ * @return false otherwise
+ * @headerfile <seqan/blast.h>
+ */
+
+constexpr bool
+qIsTranslated(BlastProgram const p)
```

Comment:
Previous comments apply to all functions below. In addition, the code below is extremely repetitive...